hare/committracker.go

```diff
@@ -59,35 +59,45 @@ func (ct *commitTracker) OnCommit(ctx context.Context, msg *Msg) {
 		return
 	}
 
+	msgHash := types.BytesToHash(msg.Message.HashBytes())
+	metadata := types.HareMetadata{
+		Layer:   msg.Layer,
+		Round:   msg.Round,
+		MsgHash: msgHash,
+	}
 	if prev, ok := ct.seenSenders[msg.NodeID]; ok {
-		if prev.InnerMsg.Layer == msg.Layer &&
-			prev.InnerMsg.Round == msg.Round &&
-			prev.InnerMsg.MsgHash != msg.MsgHash {
-			ct.logger.WithContext(ctx).With().Warning("equivocation detected in commit round",
+		if prev.InnerMsg.Layer != msg.Layer || prev.InnerMsg.Round != msg.Round {
```

Comment:
this check makes me think that mismatching layer / round is expected. but it's actually not.
mismatching layer here would be a programming error.
mismatching round here should not even be processed.
the old code already have this trouble.

Reply:
I hope you also find the `Equivocation` method a cleaner approach to detecting two messages of different content in the same layer 🙂